### PR TITLE
feat: add mjs support by add `index.mjs`

### PR DESCRIPTION
**Pull request by xiaoxiangmoe**

### CODE REVIEW

File: index.mjs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import JsoncParser from './lib/umd/main.js';
```
Comment (aeschli) [inline index.mjs:1]: can this be lib/esm/main.js?

Comment (xiaoxiangmoe) [reply index.mjs:1]: No, in `lib/esm/main.js`, we import files without `'.js'` suffix, which will cause error.

Comment (xiaoxiangmoe) [reply index.mjs:1]: https://nodejs.org/api/esm.html#esm_terminology

> Relative specifiers like './startup.js' or '../config.mjs'. They refer to a path relative to the location of the importing file.

**Comment by aeschli** [comment]
Closing. I need more help here to understand what the advantages are.

**Pull request closed without merging**